src/shared/utils.test.ts

```diff
@@ -0,0 +1,31 @@
+import { roundBinSize } from './utils';
+
+const BIN_SIZES_INTEGER = [1, 2, 3, 5, 10, 15, 20];
+const BIN_SIZES_REAL = [1, 1.25, 2, 2.5, 3, 5, 7.5, 10];
+
+test('roundBinSize', () => {
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(x, BIN_SIZES_INTEGER)).toBe(x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(x, BIN_SIZES_REAL)).toBe(x));
+
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(10 * x, BIN_SIZES_INTEGER)).toBe(10 * x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(10 * x, BIN_SIZES_REAL)).toBe(10 * x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(x / 10, BIN_SIZES_REAL)).toBe(x / 10));
+
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(1.01 * x, BIN_SIZES_INTEGER)).toBe(x));
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(0.99 * x, BIN_SIZES_INTEGER)).toBe(x));
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(1.000000001 * x, BIN_SIZES_INTEGER)).toBe(x));
+  BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(0.999999999 * x, BIN_SIZES_INTEGER)).toBe(x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(1.01 * x, BIN_SIZES_REAL)).toBe(x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(0.99 * x, BIN_SIZES_REAL)).toBe(x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(1.000000001 * x, BIN_SIZES_REAL)).toBe(x));
+  BIN_SIZES_REAL.map((x) => expect(roundBinSize(0.999999999 * x, BIN_SIZES_REAL)).toBe(x));
+
+  expect(roundBinSize(0.0000000000000001, BIN_SIZES_INTEGER)).toBe(0.0001);
+  expect(roundBinSize(0.0000000000000001, BIN_SIZES_REAL)).toBe(0.0001);
+  expect(roundBinSize(0, BIN_SIZES_INTEGER)).toBe(0.0001);
+  expect(roundBinSize(0, BIN_SIZES_REAL)).toBe(0.0001);
+  const hugeInt = 1000000000000000;
+  expect(roundBinSize(hugeInt, BIN_SIZES_INTEGER)).toBeCloseTo(hugeInt);
```

Comment:
Should be equal because given number is less than `MAX_SAFE_INTEGER` (9007199254740991)